packages/perspective-viewer-d3fc/src/js/legend/scrollableLegend.js

```diff
@@ -71,17 +73,34 @@ export default fromLegend => {
         const legendElement = getLegendElement(selection);
         legendElement.call(legend);
 
-        const cellSize = selection
+        const cellContainerSize = selection
             .select("g.legendCells")
             .node()
             .getBBox();
-        legendElement.attr("height", cellSize.height + 20);
+        legendElement.attr("height", cellContainerSize.height + 20);
 
         decorate(selection);
     };
 
-    const cellFilter = () => {
-        return (_, i) => i >= pageSize * pageIndex && i < pageSize * pageIndex + pageSize;
+    const cellFilter = () => (_, i) => i >= pageSize * pageIndex && i < pageSize * pageIndex + pageSize;
+
+    const calculatePageSize = selection => {
+        const legendContainerRect = selection.node().getBoundingClientRect();
+        const averageCellHeight = 16;
+        let proposedPageSize = Math.floor(legendContainerRect.height / averageCellHeight) - 1;
+
+        pageCount = calculatePageCount(proposedPageSize);
+
+        //if page size is less than all legend items, leave space for the legend controls
+        pageSize = proposedPageSize < domain.length ? proposedPageSize - 1 : proposedPageSize;

```

Comment:
It seems like we've potentially added an extra page by making this adjustment, but we didn't fix `pageCount`?

Reply:
Right you are. Will fix.